### PR TITLE
sql: CANCEL QUERY error tweaks, SHOW SESSIONS semicolon removal

### PR DESCRIPTION
**Pull request by itsbilal**
- Lowercase CANCEL QUERY errors to follow convention.
- Add pgcode to query cancelled error.
- Remove the semicolon at the end of `active_queries` in `SHOW SESSIONS`, to make the output match that of the `query` field in `SHOW QUERIES`.

### CODE REVIEW

File: pkg/sql/crdb_internal.go [hunk 1]
```diff
@@ -697,9 +697,12 @@ func populateSessionsTable(
 		var oldestStart time.Time
 		var oldestStartDatum parser.Datum
 
-		for _, query := range session.ActiveQueries {
+		numQueries := len(session.ActiveQueries)
+		for idx, query := range session.ActiveQueries {
```
Comment (knz) [inline pkg/sql/crdb_internal.go]:
```
for i, query := range ...{
   if i > 0 {
      WriteString("; ")
  }
   WriteString(sql)
}
```

Comment (itsbilal) [inline pkg/sql/crdb_internal.go]: Done

**Review by knz — APPROVED** [review]

**Comment by knz** [comment]
LGTM modulo clean up in test.

---

Reviewed 4 of 4 files at r1, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/sql/logictest/testdata/logic_test/show_source, line 158 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/17739#-Ks46mTylJgr5tEQctnv:-Ks46mTylJgr5tEQctnw:b-ooan0q) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b113bf2a7262ee2d3c625ec6ecd73d7cc52b832/pkg/sql/logictest/testdata/logic_test/show_source#L158)):*
> ```
> SELECT node_id,username,application_name,active_queries FROM [SHOW SESSIONS] WHERE active_queries != ''
> ----
> node_id  username  application_name  active_queries                                   
> ```

Remove the trailing whitespaces.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17739#-:-Ks46qQhDuG03xLa78mH:bgbv958)*

**Comment by itsbilal** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/sql/logictest/testdata/logic_test/show_source, line 158 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/17739#-Ks46mTylJgr5tEQctnv:-Ks4EOkwn9WN8pGhbCme:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b113bf2a7262ee2d3c625ec6ecd73d7cc52b832/pkg/sql/logictest/testdata/logic_test/show_source#L158)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Remove the trailing whitespaces.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17739)*

**Pull request merged**